### DISCUSSION TITLE
`chain` is `undefined` after `switchNetworkAsync`

### DISCUSSION THREAD

**Discussion by 0xhohenheim** [WalletConnect/walletconnect-monorepo#2555] [Developer Support]
I'm not entirely sure what conditions are required for this to occur or to be reproduced but when the network is switched on Metamask, be it manually or after calling `switchNetworkAsync`, the `chain` from `useNetwork()` updates but not always with the correct network information, sometimes it returns `undefined` which ends up breaking the UI since network warnings are dependent on this.

This occurs regardless of whether the chain was configured in the wagmi client or not. I've in-fact switched to configured chains and faced the same issue.

**Project dependencies**:
```
"react": "^18.2.0",
"react-dom": "^18.2.0",
"wagmi": "^0.12.2",
"@web3modal/ethereum": "^2.2.1",
"@web3modal/react": "^2.2.1",
"ethers": "5.6.9",
```

**Comment by xzilja** [comment]
I'm afraid there isn't much we can do here in terms of web3modal, this would be in realm of wagmi. First thing I see is that you are on a relatively outdated version of their package, so step one would be updating to latest (same for web3modal) and checking if error persists.

**Reply by 0xhohenheim** [reply]
I would lose support for ethers if I update though right? All the contract interactions would have to be rewritten.

**Comment by glitch-txs** [comment]
Could you create a minimal reproduction example? MetaMask extension right? So there no way to even caught the error?

**Reply by 0xhohenheim** [reply]
Sure, I'll set something up this week

**Comment by 0xhohenheim** [comment]
@0xAsimetriq would the above dependencies be affected by the `WalletConnect v1.0` shutdown? I would've made a separate thread to discuss this but it seemed relevant here because I'll have to migrate to the latest versions.

**Reply by xzilja** [reply]
No, but I do still recommend to upgrade to their latest versions.

**Reply by 0xhohenheim** [reply]
The project is already live on mainnet, upgrading to the latest versions would break the existing integration, it needs to be up for the next 2-3 months only so not worth it unless there's a security concern or it's affected by the shutdown. 

Will upgrade long term projects though.

Also, thanks for the quick response! 💯
